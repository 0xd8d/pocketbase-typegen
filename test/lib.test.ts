import {
	createCollectionEnum,
	createRecordType,
	createTypeField,
	generate,
} from "../src/lib"

describe("generate", () => {
	it("generates correct output given db input", () => {
		const name = "books"
		const schema = JSON.stringify([
			{
				name: "title",
				type: "text",
				required: false,
			},
		])
		const result = generate([{ name, schema }])
		expect(result).toMatchSnapshot()
	})
})

describe("createCollectionEnum", () => {
	it("creates enum of collection names", () => {
		const names = ["book", "magazine"]
		expect(createCollectionEnum(names)).toMatchSnapshot()
	})
})

describe("createRecordType", () => {
	it("creates type definition for a record", () => {
		const name = "books"
		const schema = JSON.stringify([
			{
				system: false,
				id: "hhnwjkke",
				name: "title",
				type: "text",
				required: false,
				unique: false,
				options: { min: null, max: null, pattern: "" },
			},
		])
		const result = createRecordType(name, schema)
		expect(result).toMatchSnapshot()
	})
})

describe("createTypeField", () => {
	it("handles required and optional fields", () => {
		expect(createTypeField("name", true, "text")).toEqual("\tname: string;\n")
		expect(createTypeField("name", false, "text")).toEqual("\tname?: string;\n")
	})

	it("converts pocketbase schema types to typescript", () => {
		expect(createTypeField("name", true, "text")).toEqual("\tname: string;\n")
		expect(createTypeField("textField", true, "text")).toEqual(
			"\ttextField: string;\n"
		)
		expect(createTypeField("numberField", true, "number")).toEqual(
			"\tnumberField: number;\n"
		)
		expect(createTypeField("boolField", true, "bool")).toEqual(
			"\tboolField: boolean;\n"
		)
		expect(createTypeField("emailField", true, "email")).toEqual(
			"\temailField: string;\n"
		)
		expect(createTypeField("urlField", true, "url")).toEqual(
			"\turlField: string;\n"
		)
		expect(createTypeField("dateField", true, "date")).toEqual(
			"\tdateField: string;\n"
		)
		expect(createTypeField("selectField", true, "select")).toEqual(
			"\tselectField: string;\n"
		)
		expect(createTypeField("jsonField", true, "json")).toEqual(
			"\tjsonField: string;\n"
		)
		expect(createTypeField("fileField", true, "file")).toEqual(
			"\tfileField: string;\n"
		)
		expect(createTypeField("relationField", true, "relation")).toEqual(
			"\trelationField: string;\n"
		)
		expect(createTypeField("userField", true, "user")).toEqual(
			"\tuserField: string;\n"
		)
	})
})
