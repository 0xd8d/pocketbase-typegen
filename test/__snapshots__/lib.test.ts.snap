// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createCollectionEnum creates enum of collection names 1`] = `
"export enum Collections {
	Book = "book",
	Magazine = "magazine",
}"
`;

exports[`createRecordType creates type definition for a record 1`] = `
"export type BooksRecord = {
	title?: string
}"
`;

exports[`createRecordType handles file fields with multiple files 1`] = `
"export type BooksRecord = {
	avatars?: string[]
}"
`;

exports[`generate generates correct output given db input 1`] = `
"// Generated using pocketbase-typegen

export enum Collections {
	Books = "books",
}

export type CollectionRecords = {
	books: BooksRecord
}

export type BooksRecord = {
	title?: string
}"
`;
